force-app/main/default/classes/RD2_ScheduleService.cls

```diff
@@ -1056,10 +1056,13 @@ public without sharing class RD2_ScheduleService {
         }
 
         private Boolean shouldEditPause (List<RecurringDonationSchedule__c> donationSchedules) {
+            Boolean shouldEdit = false;
             for (RecurringDonationSchedule__c schedule : donationSchedules) {
-                return schedule.IsPause__c;
+                if (schedule.IsPause__c) {
+                    shouldEdit = true;
```

Comment:
I think we can return early here the moment we find a paused schedule instead of looping through all the schedules before returning a result.

Reply:
Also thanks for jumping on this so quickly!

Reply:
Heh, I just thought of that too, I'll add a `break` since I prefer that to having multiple `return` statements.